lib/process-services-cloud/src/lib/task/task-filters/components/task-assignment-filter/task-assignment-filter.component.ts

```diff
@@ -59,35 +69,74 @@ export class TaskAssignmentFilterCloudComponent implements OnInit {
         return this.assignmentType === AssignmentType.CANDIDATE_GROUPS;
     }
 
-    onAssignmentTypeChange(type: any) {
+    isAssignedToType(): boolean {
+        return this.assignmentType === AssignmentType.ASSIGNED_TO;
+    }
+
+    onAssignmentTypeChange(assignmentChange: MatSelectChange) {
         this.candidateGroups = [];
-        if (type === AssignmentType.CURRENT_USER) {
-            this.assignedChange.emit(this.identityUserService.getCurrentUserInfo());
-        } else if (type === AssignmentType.UNASSIGNED) {
-            this.assignedChange.emit(null);
+        this.assignedUsers = [];
+
+        if (assignmentChange.value === AssignmentType.CURRENT_USER) {
+            this.assignedUsersChange.emit([this.identityUserService.getCurrentUserInfo()]);
+        } else if (assignmentChange.value === AssignmentType.NONE) {
+            this.assignedUsersChange.emit(undefined);
         }
+
+        this.assignmentType = assignmentChange.value;
+        this.assignmentTypeChange.emit(this.assignmentType);
     }
 
     onChangedGroups(groups: IdentityGroupModel[]) {
-        this.assignedGroupChange.emit(groups);
+        this.assignedGroupsChange.emit(groups);
     }
 
-    private setDefaultAssignmentType() {
-        const assignmentAttr = this.taskFilterProperty.attributes['assignee'];
-        const assignee = this.taskFilterProperty.value[assignmentAttr];
+    onChangedAssignedUsers(users: IdentityUserModel[]) {
+        this.assignedUsersChange.emit(users);
+    }
+
+    private changeAssignmentTypeByStatus(status: string) {
```

Comment:
is there any enum about the status that we could use instead of string?

Reply:
used enum instead of string